va/va_enc_hevc.h

```diff
@@ -45,6 +45,200 @@ extern "C" {
  * @{
  */
 
+/** Attribute value for VAConfigAttribEncHEVCFeatures.
+ *
+ * This attribute decribes the supported features of an HEVC/H.265
+ * encoder configuration.
+ *
+ * All of the field values in this attribute are VA_FEATURE_* values,
+ * indicating support for the corresponding feature.
+ */
+typedef union VAConfigAttribValEncHEVCFeatures {
+    struct {
+        /** Separate colour planes.
+         *
+         * Allows setting separate_colour_plane_flag in the SPS.
+         */
+        uint32_t separate_colour_planes     : 2;
+        /** Scaling lists.
+         *
+         * Allows scaling_list() elements to be present in both the SPS
+         * and the PPS.  The decoded form of the scaling lists must also
+         * be supplied in a VAQMatrixBufferHEVC buffer when scaling lists
+         * are enabled.
+         */
+        uint32_t scaling_lists              : 2;
+        /** Asymmetric motion partitions.
+         *
+         * Allows setting amp_enabled_flag in the SPS.
+         */
+        uint32_t amp                        : 2;
+        /** Sample adaptive offset filter.
+         *
+         * Allows setting slice_sao_luma_flag and slice_sao_chroma_flag
+         * in slice headers.
+         */
+        uint32_t sao                        : 2;
```

Comment:
Driver allows these two slice_sao_luma_flag and slice_sao_chroma_flag to be disabled/enabled respectively. How to set this field, which bit represent luma/chroma?

Reply:
Are you thinking of hardware which supports SAO only on one type of plane but not the other?  I was thinking that no such hardware would exist, but if you think it is possible then we could add separate `sao_luma` and `sao_chroma` fields to distinguish the cases.

Reply:
Rereading that, I'm now not sure that that was your question.  To clarify my intended meaning:

If `VAConfigAttribValEncHEVCFeatures.sao == VA_FEATURE_NOT_SUPPORTED`, then the user must set both `slice_sao_luma_flag` and `slice_sao_chroma_flag` to zero.

If `VAConfigAttribValEncHEVCFeatures.sao == VA_FEATURE_SUPPORTED`, then the user can pick whatever values they want for `slice_sao_luma_flag` and `slice_sao_chroma_flag`, and they need not be the same.

If `VAConfigAttribValEncHEVCFeatures.sao == VA_FEATURE_REQUIRED`, then the user must set both `slice_sao_luma_flag` and `slice_sao_chroma_flag` to one.

Do you have/expect hardware for which that would be a problem?